script_armory/client.lua

```diff
@@ -8,111 +8,107 @@
-                       end
-                       if CheckPos(coords.x, coords.y, coords.z, g1.x, g1.y, g1.z, 2) then 
-                               alert(Config.Message)
-                               if IsControlJustPressed(1, Config.Keybind) then 
-                              RemoveAllPedWeapons(GetPlayerPed(-1), true)
-
-                              for _,weapon in ipairs(Config.Weapons) do
-                                giveWeapon(weapon)
-
-
-                              end
-
-                              for _,comp in ipairs(Config.Components) do
-                                weaponComponent(comp[1], comp[2])
-                               
+        Citizen.Wait(200)
+
+        local ped = PlayerPedId()
+        local coords = GetEntityCoords(ped)
+
+        local lowestDist = Config.Marker.DrawDistance
+        for _,armory in pairs(Config.Armories) do
+            local distance = GetDistanceBetweenCoords(coords, armory.x, armory.y, armory.z, true)
+            if distance < lowestDist then
+                closestArmory.loc = armory
+                lowestDist = distance
+
+                if distance < Config.Marker.Radius then
+                    closestArmory.isInMarker = true
+                else
+                    closestArmory.isInMarker = false
+                end
+            end
+        end
+
+        if lowestDist == Config.Marker.DrawDistance then
+            closestArmory.loc = {}
+        end
+    end
+end)
 
-                              end
-                        if Config.armor then
-                              AddArmourToPed(GetPlayerPed(-1), 100)
-                        end
+Citizen.CreateThread(function()
+    while true do
+        Citizen.Wait(0)
 
+        if closestArmory.loc ~= {} then
+            local markerScale = Config.Marker.Scale
+            local markerColor = Config.Marker.Color
 
+            DrawMarker(1, closestArmory.loc.x, closestArmory.loc.y, closestArmory.loc.z, 0, 0, 0, 0, 0, 0, markerScale.x, markerScale.y, markerScale.z, markerColor.r, markerColor.g, markerColor.b, markerColor.a, false, false, 2, false, nil, nil, false)
 
-                              ShowNotification("~g~Police loadout has been loaded.")
-                            
-                            end
-                       end
-                   
-               end
-           end
-       
-       
-   end)
+            if closestArmory.isInMarker then
+                ShowAlert(Config.Message)
 
+                if IsControlJustPressed(0, Config.Keybind) then
+                    local ped = PlayerPedId()
 
+                    RemoveAllPedWeapons(ped, true)
 
+                    for _,weapon in ipairs(Config.Weapons) do
+                        giveWeapon(weapon)
+                    end
 
+                    for _,comp in ipairs(Config.Components) do
+                        weaponComponent(comp.weapon, comp.component)

```

Comment:
better to send ped from here and don't call to get player inside of  function for optimizing
```lua
weaponComponent(ped, comp.weapon, comp.component)
```